### PR TITLE
ls: cross-platform colors

### PR DESCRIPTION
**Pull request by tertsdiepraam**
Previously, the colors in `ls` were restricted to Unix. This PR removes that restriction and uses the [`lscolors`](https://github.com/sharkdp/lscolors) crate instead of a custom implementation.

### CODE REVIEW

**Comment by sylvestre** [comment]
Plenty of code removed, the best kind of PR, bravo :)
There is a build failure but I am sure you know this!

**Comment by tertsdiepraam** [comment]
@sylvestre Thanks, I hadn't seen that yet! It should be fixed now! One Ubuntu and FreeBSD still fail, but for unrelated reasons.

File: src/uu/ls/src/ls.rs [hunk 1]
```diff
@@ -1626,23 +1522,29 @@ fn display_file_name(
...
 
     if config.format == Format::Long && metadata.file_type().is_symlink() {
         if let Ok(target) = path.read_link() {
-            // We don't bother updating width here because it's not used for long listings
-            let code = if target.exists() { "fi" } else { "mi" };
-            let target_name = color_name(target.to_string_lossy().to_string(), code);
+            // We don't bother updating width here because it's not used for long
+            let mut target_name = target.to_string_lossy().to_string();
+            if let Some(ls_colors) = &config.color {
+                target_name = color_name(&ls_colors, &target, target_name, metadata);
```
Comment (sylvestre) [inline src/uu/ls/src/ls.rs:1533]: would it be possible to have a test covering this line?

Comment (tertsdiepraam) [reply src/uu/ls/src/ls.rs:1533]: Ah interesting, I just noticed that GNU doesn't actually color the target, so I guess I'll just remove that line:
image

**Comment by tertsdiepraam** [comment]
It should be all good now!

**Comment by sylvestre** [comment]
Great work, bravo :)

**Pull request merged**